File: server_integration_tests/servers/java_server/src/main/java/dev/graphlink/javaserver/impl/DeleteArticleServiceImpl.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.services.DeleteArticleService;
import org.springframework.stereotype.Service;

@Service
public class DeleteArticleServiceImpl implements DeleteArticleService {

    @Override
    public boolean deleteArticle(String id) {
        return Data.articles.removeIf(a -> a.getId().equals(id));
    }
}

File: server_integration_tests/kotlin/kotlin_server_blocking/src/main/kotlin/dev/graphlink/kotlinserverblocking/impl/DeleteArticleServiceImpl.kt
package dev.graphlink.kotlinserverblocking.impl

import dev.graphlink.kotlinserverblocking.generated.services.DeleteArticleService
import org.springframework.stereotype.Service

@Service
class DeleteArticleServiceImpl : DeleteArticleService {
    override fun deleteArticle(id: String): Boolean = Data.articles.removeAll { it.id == id }
}
